test/e2e/persistent_volumes.go

```diff
@@ -66,58 +75,102 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
 			persistentVolumeTestCleanup(c, config)
 		}()
 
+		// Create the nfs server pod
 		pod := startVolumeServer(c, config)
 		serverIP := pod.Status.PodIP
 		framework.Logf("NFS server IP address: %v", serverIP)
 
+		// Define the PersistentVolume and PersistentVolumeClaim
 		pv := makePersistentVolume(serverIP)
 		pvc := makePersistentVolumeClaim(ns)
 
-		framework.Logf("Creating PersistentVolume using NFS")
+		// Create the PersistentVolume and wait for PersistentVolume.Status.Phase to be Available
+		// defer deletion to clean up the PV should the test fail post-creation.
+		framework.Logf("Creating PersistentVolume")
 		pv, err := c.PersistentVolumes().Create(pv)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			framework.Failf("Create PersistentVolume failed: %v", err)
+		}
+		defer deletePersistentVolume(c, pv)
+		framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 1*time.Second, 20*time.Second)
 
+		// Create the PersistentVolumeClaim and wait for Bound phase
 		framework.Logf("Creating PersistentVolumeClaim")
 		pvc, err = c.PersistentVolumeClaims(ns).Create(pvc)
-		Expect(err).NotTo(HaveOccurred())
-
-		// allow the binder a chance to catch up.  should not be more than 20s.
-		framework.WaitForPersistentVolumePhase(api.VolumeBound, c, pv.Name, 1*time.Second, 30*time.Second)
+		if err != nil {
+			framework.Failf("Create PersistentVolumeClaim failed: %v", err)
+		}
+		framework.WaitForPersistentVolumeClaimPhase(api.ClaimBound, c, ns, pvc.Name, 1*time.Second, 20*time.Second)
```

Comment:
`WaitForPersistentVolumeClaimPhase` should have the same timeout as `WaitForPersistentVolumePhase` below.